dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-etcd/src/main/java/org/apache/dolphinscheduler/plugin/registry/etcd/EtcdRegistry.java

```diff
@@ -0,0 +1,344 @@
+                for (WatchEvent event : watchResponse.getEvents()) {
+                    listener.notify(new EventAdaptor(event, path));
+                }
+            }));
+        } catch (Exception e) {
+            throw new RegistryException("Failed to subscribe listener for key: " + path, e);
+        }
+        return true;
+    }
+
+    /**
+     * @throws throws an exception if the unsubscribe path does not exist
+     * @param path The prefix of the key being listened to
+     */
+    @Override
+    public void unsubscribe(String path) {
+        try {
+            watcherMap.get(path).close();
+            watcherMap.remove(path);
+        } catch (Exception e) {
+            throw new RegistryException("Failed to unsubscribe listener for key: " + path, e);
+        }
+    }
+
+    @Override
+    public void addConnectionStateListener(ConnectionListener listener) {
+        etcdConnectionStateListener.addConnectionListener(listener);
+    }
+
+    /**
+     *
+     * @param key
+     * @return Returns the value corresponding to the key
+     * @throws throws an exception if the key does not exist
+     */
+    @Override
+    public String get(String key) {
+        try {
+            List<KeyValue> keyValues = client.getKVClient().get(byteSequence(key)).get().getKvs();
+            return keyValues.iterator().next().getValue().toString(StandardCharsets.UTF_8);
+        } catch (InterruptedException | ExecutionException e) {
+            Thread.currentThread().interrupt();
+            throw new RegistryException("etcd get data error", e);
+        }
+    }
+
+    /**
+     *
+     * @param key
+     * @param value
+     * @param deleteOnDisconnect Does the put data disappear when the client disconnects
+     */
+    @Override
+    public void put(String key, String value, boolean deleteOnDisconnect) {
+        try {
+            if (deleteOnDisconnect) {
+                // keep the key by lease, if disconnected, the lease will expire and the key will delete
+                long leaseId = client.getLeaseClient().grant(TIME_TO_LIVE_SECONDS).get().getID();
+                client.getLeaseClient().keepAlive(leaseId, Observers.observer(response -> {
+                }));
+                PutOption putOption = PutOption.newBuilder().withLeaseId(leaseId).build();
+                client.getKVClient().put(byteSequence(key), byteSequence(value),putOption).get();
+            } else {
+                client.getKVClient().put(byteSequence(key), byteSequence(value)).get();
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            Thread.currentThread().interrupt();
+            throw new RegistryException("Failed to put registry key: " + key, e);
+        }
+    }
+
+    /**
+     * delete all keys that contain the prefix
+     * @param key the prrefix
+     */
+    @Override
+    public void delete(String key) {
```

Comment:
It's better to use the more precise method name, such as `deleteByPrefix(String keyPrefix)`

Reply:
> It's better to use the more precise method name, such as `deleteByPrefix(String keyPrefix)`

This is a function from the Registry interface. It may be a better choice to initiate a new PR to modify the function name.

Reply:
This is a function from the Registry interface, maybe we can start an Issue to see other people's opinions

Reply:
I also think deleteByPrefix can better reflect the meaning of the function